dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -324,6 +357,37 @@ public void Dispose()
         return result;
     }
 
+    /// <summary>
+    /// Execute the FunctionInvoking event handlers and returns true if no cancellation was attempted.
+    /// </summary>
+    /// <param name="functionView">Function view details</param>
+    /// <param name="context">SKContext before function invocation</param>
+    /// <param name="renderedPrompt">Rendered prompt prior to semantic function invocation</param>
+    /// <returns>Returns true if no cancellation was attempted.</returns>
+    private bool OnFunctionInvoking(FunctionView functionView, SKContext context, string? renderedPrompt)
+    {
+        if (this.FunctionInvoking is not null)
+        {
+            var args = new FunctionInvokingEventArgs(functionView, context, renderedPrompt);
+            this.FunctionInvoking.Invoke(this, args);
```

Comment:
Initially I considered this approach where the Cancel would abort further callers.

As an alternative I'm adding two new properties to `FunctionInvokingEventArgs`:

Added `Skip` will skip current Function, moving to the next skill in the pipeline/plan. 
Renamed `Cancel` to `Abort` will stop all execution.

For both properties, once set to true by one handler the values cannot be unset by any further  handler.
Any further attempt to unset it will throw.